### PR TITLE
fix: next_event can now process multiple sequential events

### PR DESCRIPTION
**Pull request by itayd**
there was a problem where next_event skipped events. this change the name of the last event read in sessions to be more explicit, as well as the condition in the db to be more appropriate with the filter meaning. also increase the seq number in the filter given to List as needed.

### CODE REVIEW

File: internal/backend/sessions/sessionworkflows/workflow.go [hunk 1]
```diff
@@ -111,7 +111,7 @@ func runWorkflow(
 
 func (w *sessionWorkflow) cleanupSignals(ctx workflow.Context) {
 	goCtx := temporalclient.NewWorkflowContextAsGOContext(ctx)
-	for signalID := range w.signals {
+	for signalID := range w.lastReadEventSeqForSignal {
```
Comment (efiShtain) [inline internal/backend/sessions/sessionworkflows/workflow.go:114]: I like the rename for readability, but it makes this part is now less readable

File: internal/backend/sessions/sessionworkflows/workflow.go [hunk 2]
```diff
@@ -378,16 +378,16 @@ func (w *sessionWorkflow) getNextEvent(ctx context.Context, signalID string) (ma
...
-		w.z.Panic("signal not found")
+		return nil, fmt.Errorf("no such subscription %q", signalID)
 	}
 
 	filter := sdkservices.ListEventsFilter{
 		IntegrationID:     iid,
 		ConnectionID:      cid,
 		Limit:             1,
-		MinSequenceNumber: minSequenceNumber,
+		MinSequenceNumber: minSequenceNumber + 1,
```
Comment (efiShtain) [inline internal/backend/sessions/sessionworkflows/workflow.go:387]: I'm not sure why +1 here and >= in the db layer makes a difference 

it's suppose to be the same as minSequenceNumber and >

**Review by efiShtain — APPROVED** [review]
not sure how it works, and a little name that bugs me, but looks ok

**Comment by efiShtain** [comment]
just fix the bugs before you try to merge please

File: internal/backend/sessions/sessionworkflows/workflow.go (hunk 2 above)
Comment (itayd) [reply internal/backend/sessions/sessionworkflows/workflow.go:387]: if it's min AND >, it'll not include the min. I want it to include the min.

File: internal/backend/sessions/sessionworkflows/workflow.go (hunk 1 above)
Comment (itayd) [reply internal/backend/sessions/sessionworkflows/workflow.go:114]: that's the most meaningful name i could come up with.

**Pull request merged**